### DISCUSSION TITLE
Unable to access elements of ListCoursesResponse

### DISCUSSION THREAD

**Discussion by pmnlla** [Byron/google-apis-rs#449] [Q&A] [answered]
Heyo! I'm trying to access the course list of `hub.courses().list().doit().await;` where I assume that each course is a separate `Course` struct, through the following piece of code:

```rust
    match req {
        Err(e) => print!("{}",e),
        Ok(output) => courseListParse(output.1, clist),
    }
 
 // outside of function...
 
 fn courseListParse(mut response:ListCoursesResponse, mut clist:Vec<class::classType>){
    for i in &response.courses{
        println!("{:?}\n", i); // this print statement is a placeholder so i can view the data returned by the gclassroom api, and later aggregate it
    }
}
 
 ```
 
 however, i am running into an issue where the data returned by `response.courses` is a single element, probably a vector of classes. I cannot access any elements of this vector through this current function.
 
 I'm quite new to rust, so if I'm making any obvious mistake, please let me know. I did look over rustdoc and found out that the `courses` variable in `ListCoursesResponse` is a vector of `Course`s, which I think would typically be accessible through `response.courses.[0-x]` as per a typical vector.

**Comment by Byron** [comment]
I'd rely on auto-complete to show what's accessible in a particular struct, and in lieu of that use the docs.rs documentation of the crate.
Without the name of it I can't link to it though.

Since it seems to debug print, I'd assume it's possible to access `Course` fields as well if these are pubic. If not, that would be unusual.

**Reply by pmnlla** [reply]
Sorry I forgot to include that information. The crate is `google_classroom1`. The confusingly named second variable in `courseListParse`, `"clist`, is of a custom structure I built so I wouldn't have to parse the api response every time. You can ignore that.

This is the autocomplete response for `response.courses`:
![image](https://github.com/Byron/google-apis-rs/assets/39027165/129d16cc-fa41-4aa8-9262-63bd40b4f6c0)

This is the autocomplete response for `i`:
![image](https://github.com/Byron/google-apis-rs/assets/39027165/f3548b25-cde9-4004-80c8-a9edc9cc3e72)

**Reply by Byron** [reply, marked as answer]
I'd recommend to [look up the docs](https://docs.rs/google-classroom1/latest/google_classroom1/api/struct.ListCoursesResponse.html) and see how the type `Option<Vec<Course>>` can be handled. What's probably confusing here is that `Option` also can be iterated, which is why that loop seems to return a `Vec<Course>` itself, which of course could now be iterated as well.

Since this is a Rust question, it's not at all related to this crate as initially suggested.

**Reply by pmnlla** [reply]
I didn't realize that `Option` could be iterated through. Thank you so much! I'll have a shot at passing the response again when I'm home.
